### DISCUSSION TITLE
Ory ssh key authentication possible/reasonable?

### DISCUSSION THREAD

**Discussion by appst** [ory/kratos#2453] [Ideas]
Recently I encountered a project that uses ssh keys for authentication in a way I have not seen before: https://github.com/charmbracelet/wish

I become instantly intrigued when I ran their example in my terminal...
ssh git.charm.sh

This example makes me wonder if terminal apps could become first-class citizens.

It seems to me it would go like this...
The host server at git.charm.sh looks for the user's public key record within its publickey database.
If there is an entry then the server knows who the user is and can proceed accordingly.
If there is no such public key record then the server can prompt the user to create an account.
Prompting the user for SSH key signin/signup will be done through the user's terminasl using charmbracelet/wish methods.

Web authorization would remain unchanged, and a user could manage their SSH keys through standard web methods.

Thoughts?

**Comment by Benehiko** [comment]
Hi @appst 

The idea sounds quite good, however, Kratos doesn't support creating sessions from e.g. an admin endpoint. Since in this case you would be doing the authentication through a 3rd party service. Am I correct in assuming this?

I would suggest opening an issue in Kratos for supporting such a method - but I'm not sure if this would be out of scope of the project.
